libs/config/settings.c

```diff
@@ -435,6 +435,88 @@ static int settings_parse(JsonNode *root) {
 			} else {
 				settings_add_string(jsettings->key, jsettings->string_);
 			}
+		} else if(strcmp(jsettings->key, "mail-from") == 0 ||
+					strcmp(jsettings->key, "mail-to") == 0 || 
+					strcmp(jsettings->key, "smtp-user") == 0) {
+			if(jsettings->tag != JSON_STRING) {
+				logprintf(LOG_ERR, "config setting \"%s\" must contain an e-mail address", jsettings->key);
+				have_error = 1;
+				goto clear;
+			} else if(!jsettings->string_) {
+				logprintf(LOG_ERR, "config setting \"%s\" must contain an e-mail address", jsettings->key);
+				have_error = 1;
+				goto clear;
+			} else if(strlen(jsettings->string_) > 0) {
+				char validate[] = "^[a-zA-Z0-9_.]+@([a-zA-Z0-9]+.)+[a-zA-Z0-9]{2,3}$";
+				reti = regcomp(&regex, validate, REG_EXTENDED);
+				if(reti) {
+					logprintf(LOG_ERR, "could not compile regex for %s", jsettings->key);
+					have_error = 1;
+					goto clear;
+				}
+				reti = regexec(&regex, jsettings->string_, 0, NULL, 0);
+				if(reti == REG_NOMATCH || reti != 0) {
+					logprintf(LOG_ERR, "config setting \"%s\" must contain an e-mail address", jsettings->key);
+					have_error = 1;
+					regfree(&regex);
+					goto clear;	
+				} else {
+					settings_add_string(jsettings->key, jsettings->string_);
+				}
+			}
+		} else if(strcmp(jsettings->key, "mail-subject") == 0 ||
+					strcmp(jsettings->key, "smtp-password") == 0) {
+			if(jsettings->tag != JSON_STRING) {
+				logprintf(LOG_ERR, "config setting \"%s\" must contain a string", jsettings->key);
+				have_error = 1;
+				goto clear;
+			} else if(!jsettings->string_) {
```

Comment:
Please use `== NULL` here (and add the specific value we are comparing to everywhere.